### DISCUSSION TITLE
The test loss is much larger than the train loss.

### DISCUSSION THREAD

**Discussion by ehsanngh** [lululxvi/deepxde#958] [Q&A]
Hello,

I'm trying to solve this first order ordinary differential equation:
$C \dfrac{dP}{dt} + \dfrac{P}{R_2} = \left(1 + \dfrac{R_1}{R_2}\right) I + R_1 C \dfrac{dI}{dt}$

$I(t)$ and $P(t)$ are functions of time. I use the following function for $I(t)$:
<img src="https://user-images.githubusercontent.com/59973930/194360433-2f75ff0a-f391-4897-94b4-910cc0caa0ef.png" width="250">

By knowing $I(t)$, I want to find $P(t)$. My problem is that there is a huge difference between the training and test losses, although their plots are pretty close to each other.
```
95000     [1.58e+00, 4.28e-04]    [3.48e+01, 4.28e-04]    [2.61e-02]    
96000     [1.68e-01, 2.60e-04]    [3.98e+01, 2.60e-04]    [2.11e-02]    
97000     [4.21e-01, 3.27e-04]    [4.13e+01, 3.27e-04]    [2.43e-02]    
98000     [4.80e-03, 2.53e-04]    [3.84e+01, 2.53e-04]    [2.00e-02]    
99000     [2.29e-02, 2.70e-04]    [3.71e+01, 2.70e-04]    [2.09e-02]    
100000    [5.68e-01, 3.62e-04]    [4.33e+01, 3.62e-04]    [2.34e-02]    

Best model at step 98000:
  train loss: 5.05e-03
  test loss: 3.84e+01
  test metric: [2.00e-02]
```
![loss (1)](https://user-images.githubusercontent.com/59973930/194363470-0e46881e-ce2a-4352-87af-9405a4a75850.png)
![prediction1](https://user-images.githubusercontent.com/59973930/194363477-e92ae3e3-02ef-422b-827d-751ea159c8cc.png)

Do you have any suggestion for me? In addition, I cannot understand how the  test error is calculated for the ODE.
Thanks for your time.

**Comment by lululxvi** [comment]
This could be due to a lot of reasons. You need to investigate carefully.

**Reply by ehsanngh** [reply]
Many thanks for your answer. Could you please tell me where I can find out about the test loss? I still have difficulties here. When we do not know the analytical solution, the train and the test loss are equal which means that the same set of collocation points is used for both the train and test. Is it right? Then, if we know the analytical solution, the deepXDE calculates the derivative of this analytical solution to find the test loss. Is it right? If so, how the derivatives of an analytical solution are computed? Thanks again. I sincerely appreciate your help.

**Reply by lululxvi** [reply]
See FAQ "Q: What is the output of DeepXDE? How can I visualize the results?"

You seem to miss some basic usage of DeepXDE. I suggest reading the demos first.
